File: tools/project-memory-context/tests/fixtures/sample.cpp
#include <string>

namespace Animals {

class Animal {
public:
    virtual std::string speak() = 0;
    void breathe() {}
};

struct Point {
    double x;
    double y;
};

} // namespace Animals

int standaloneFunction(int x, int y) {
    return x + y;
}

void helperVoid(const std::string& msg) {}

File: tools/project-memory-context/tests/fixtures/sample.py
class Animal:
    def speak(self):
        pass

class Dog(Animal):
    def speak(self):
        return "woof"

def standalone_function(x, y):
    return x + y

async def async_handler(request):
    pass

def _private_function():
    pass

@staticmethod
def decorated_func():
    pass
